Add spec details based on test plan review feedback (#8169)
dotnet/csharplang: proposals/partial-properties.md

    - Exception: this does not apply to the `extern` modifier, which may only appear on an *implementing declaration*.
4. All other syntactic differences in the signatures of partial property declarations result in a compile-time warning, with the following exceptions:
    - Attribute lists on or within partial property declarations do not need to match. Instead, merging of attributes in corresponding positions is performed, per [Attribute merging](#attribute-merging).
    - Nullable context differences do not cause warnings. In other words, a difference where one of the types is nullable-oblivious and the other type is either nullable-annotated or not-nullable-annotated does not result in any warnings.
    - Default parameter values do not need to match. A warning is reported when the implementation part of a partial indexer has default parameter values. This is similar to an existing warning which occurs when the implementation part of a partial method has default parameter values.
5. A warning occurs when parameter names differ across defining and implementing declarations. The parameter names from the definition part are used at use sites and in emit.

> 6. Nullability differences which do not involve oblivious nullability result in warnings. When analyzing an accessor body, the implementation part signature is used. The definition part signature is used when analyzing use sites and in emit. This is consistent with partial methods.

Author: updated this language per @cston feedback

> 6. Nullability differences which do not involve oblivious nullability result in warnings. When analyzing an accessor body, the parameter nullability from the definition part is used, and the return type nullability from the implementation part is used. Only the nullability of the definition part is used at use sites and in emit. This is consistent with partial methods.

Reviewer: I've noticed you can have partial methods which differ in dynamic/object parameter types and there is no warning ([SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgBwgJwC5QgGwAQAEBM2AwgLABQA3udjXgMzZJoY6yrYCyAjABQAmATxgQwUAMbYAHgEoA3NVq4GTdFmxtOvAPbAAVgFMx7AdOwBeAHzYu8srWwKaSxilWsY7Dvh46DRqXKO9C7Mahpe/EIi4tgm5lb4tgC+QA==)), but looks like there is a warning with partial properties/indexers. Is that intended?

Author: Sharplab configuration doesn't have the warning wave enabled. The warning for this difference in methods should show up in a net8 console app for example.

> 6. Nullability differences which do not involve oblivious nullability result in warnings. When analyzing an accessor body, the parameter nullability from the definition part is used, and the return type nullability from the implementation part is used. Only the nullability of the definition part is used at use sites and in emit. This is consistent with partial methods.

Author: ```suggestion
6. Nullability differences which do not involve oblivious nullability result in warnings. When analyzing an accessor body, the implementation part signature is used. The definition part signature is used when analyzing use sites and in emit. This is consistent with partial methods.
```
